src/api/photos/controller.ts

```diff
@@ -159,7 +159,12 @@ export class PhotosController {
       }
       throw err;
     }
-    
+
+    try {
+      this.usersUsecase.updateGalleryUsage(createdPhoto.userId, createdPhoto.size);
+    } catch (err) {
```

Comment:
This is incorrect as you are not awaiting the call, therefore the context is lost and the catch does not work. Use .catch() after the method call.